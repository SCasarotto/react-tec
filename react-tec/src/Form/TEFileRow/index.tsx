import React from 'react'

import { TERow, TERowCustomProps } from '../TERow'
import { TELabel, TELabelCustomProps } from '../TELabel'
import { TEFileInput, TEFileInputProps } from '../TEFileInput'

export interface TEFileRowProps
	extends TERowCustomProps,
		TELabelCustomProps,
		TEFileInputProps {
	labelForKey: string
}
export const TEFileRow: React.FC<TEFileRowProps> = (props) => {
	const { rowSize, last, className = '', title, labelForKey, ...rest } = props

	return (
		<TERow
			rowSize={rowSize}
			last={last}
			className={`TEFileRow ${className}`}>
			<TELabel
				htmlFor={labelForKey}
				required={rest.required}
				disabled={rest.disabled}
				className="TEFileRowTitle">
				{title}
			</TELabel>
			<TEFileInput
				id={labelForKey}
				className="TEFileRowInput"
				{...rest}
			/>
		</TERow>
	)
}
