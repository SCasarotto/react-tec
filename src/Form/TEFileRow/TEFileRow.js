import React from 'react'
import PropTypes from 'prop-types'

import TERow from './../TERow'
import TELabel from './../TELabel'
import TEFileInput from './../TEFileInput'

const TEFileRow = (props) => {
	const {
		//Row
		size,
		last,
		className,

		//Label
		title,

		//Both
		labelForKey,

		...rest
	} = props

	return (
		<TERow size={size} last={last} className={className}>
			<TELabel
				htmlFor={labelForKey}
				required={rest.required}
				disabled={rest.disabled}
				className='TEFileRowTitle'
			>
				{title}
			</TELabel>
			<TEFileInput id={labelForKey} className='TEFileRowInput' {...rest} />
		</TERow>
	)
}

TEFileRow.propTypes = {
	//Row
	size: PropTypes.string,
	last: PropTypes.bool,
	title: PropTypes.string,

	//Both
	labelForKey: PropTypes.string,
}

export default TEFileRow
