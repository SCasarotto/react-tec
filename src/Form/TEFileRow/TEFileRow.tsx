import React from 'react'

import { TERow, TERowCustomProps } from '../TERow'
import { TELabel, TELabelCustomProps } from '../TELabel'
import { TEFileInput, TEFileInputProps } from '../TEFileInput'

export interface TEFileRowProps extends TERowCustomProps, TELabelCustomProps, TEFileInputProps {
	labelForKey: string
}
/**
 *
 * A form row with title and `TEFileRow`. In additiona to the props below, reference all props for `TEFileRow`.
 *
 */
export const TEFileRow = React.forwardRef<HTMLInputElement, TEFileRowProps>((props, ref) => {
	const { rowSize, last, className = '', title, labelForKey, ...rest } = props

	return (
		<TERow rowSize={rowSize} last={last} className={`TEFileRow ${className}`}>
			{title && (
				<TELabel
					htmlFor={labelForKey}
					required={rest.required}
					disabled={rest.disabled}
					className='TEFileRowTitle'
				>
					{title}
				</TELabel>
			)}
			<TEFileInput id={labelForKey} className='TEFileRowInput' ref={ref} {...rest} />
		</TERow>
	)
})
