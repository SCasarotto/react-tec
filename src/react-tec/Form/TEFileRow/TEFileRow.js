//
//TE Version 0.3.0
//

import React, { useState } from 'react'
import PropTypes from 'prop-types'

import TERow from './../TERow'
import TELabel from './../TELabel'
import TEFileInput from './../TEFileInput'

//import {} from './styledComponents'

const TEFileRow = (props) => {
    const [inputKey, setInputKey] = useState('inputKey')

    const handleFileClick = (event) => {
        setInputKey(new Date().getTime())
    }

    const {
        //Row
        size,
        last,
        className,

        //Label
        title,

        //Input
        onChange,
        accept,
        multiple,
        disabled,

        //Both
        required,
        labelForKey,

        ...rest
    } = props

    return (
        <TERow size={size} last={last} className={className}>
            <TELabel
                htmlFor={labelForKey}
                labelText={title}
                required={required}
                disabled={disabled}
            />
            <TEFileInput
                resetKey={inputKey} //Used to reset the input.
                onChange={onChange}
                accept={accept}
                onClick={handleFileClick}
                id={labelForKey}
                multiple={multiple}
                disabled={disabled}
                {...rest}
            />
        </TERow>
    )
}

TEFileRow.propTypes = {
    //Row
    rowStyles: PropTypes.object,
    size: PropTypes.string,
    last: PropTypes.bool,

    //Input
    inputStyles: PropTypes.object,
    onChange: PropTypes.func,
    accept: PropTypes.string,
    multiple: PropTypes.bool,
    onChangeKey: PropTypes.string,
    disabled: PropTypes.bool,

    //Both
    labelForKey: PropTypes.string,
}

TEFileRow.defaultProps = {}

export default TEFileRow
