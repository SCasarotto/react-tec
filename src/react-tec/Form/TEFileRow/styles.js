import { colors } from './../../../config/styles';

export default {

}
