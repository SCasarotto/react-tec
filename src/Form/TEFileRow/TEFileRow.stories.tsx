import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEFileRow, TEFileRowProps } from './TEFileRow'

export default {
	title: 'Form/TEFileRow',
	component: TEFileRow,
	argTypes: {},
} as Meta

const Template: Story<TEFileRowProps> = (args) => <TEFileRow {...args} />

export const Default = Template.bind({})
Default.args = {
	labelForKey: 'fileRow',
	title: 'File Row',
}
