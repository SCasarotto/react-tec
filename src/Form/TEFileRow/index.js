import TEFileRow from './TEFileRow'

export default TEFileRow
