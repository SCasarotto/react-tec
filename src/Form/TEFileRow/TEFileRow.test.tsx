import React from 'react'
import ReactDOM from 'react-dom'
import { TEFileRow } from './TEFileRow'

describe('TEFileRow', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<TEFileRow labelForKey='testTEFileRow' />, div)
		ReactDOM.unmountComponentAtNode(div)
	})
})
