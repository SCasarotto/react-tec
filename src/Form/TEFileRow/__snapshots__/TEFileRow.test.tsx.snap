// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEFileRow matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  margin-bottom: 10px;
  text-align: left;
  box-sizing: border-box;
  padding-right: 0px;
  width: 100%;
}

.c6 {
  position: relative;
  display: block;
  width: 100%;
  font-size: 18px;
  padding: 5px 20px;
  text-align: center;
  cursor: pointer;
  border-radius: 5px;
  box-sizing: border-box;
  margin: 0px;
  border: 1px solid;
  -webkit-transition: color 0.2s ease-in,background-color 0.2s ease-in,border 0.2s ease-in;
  transition: color 0.2s ease-in,background-color 0.2s ease-in,border 0.2s ease-in;
}

.c6:hover,
.c6:active {
  border: 1px solid;
}

.c6:disabled {
  cursor: not-allowed;
  border: 1px solid;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-radius: 5px;
  padding-right: 5px;
  border: 1px solid;
  box-shadow: none;
  -webkit-transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
  transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
}

.c2:hover,
.c2:active {
  border: 1px solid;
  box-shadow: 0 0 0 1px inset;
}

.c3 {
  display: block;
  width: 100%;
  font-size: 14px;
  padding: 10px;
  cursor: pointer;
}

.c4 {
  margin-right: 10px;
  vertical-align: middle;
  font-size: 18px;
}

.c5 {
  display: none;
}

.c7 {
  width: auto;
  padding: 5px 10px;
  text-align: center;
}

.c8 {
  color: inherit;
  display: block;
}

<div
    class="c0 TERow TEFileRow "
  >
    <div
      class="c1 TEFileInput TEFileRowInput"
    >
      <div
        class="c2 TEFileInputInputWrapper"
      >
        <label
          class="c3 TEFileInputLabel"
        >
          <span
            class="TEFileInputCopy"
          >
            <svg
              class="c4 TEFileInputIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z"
              />
            </svg>
            Choose a file...
          </span>
          <input
            class="c5 TEFileInputInput"
            id="testTEFileRow"
            type="file"
          />
        </label>
        <button
          class="c6 TEButton c7 TEFileInputClearButton"
          type="button"
        >
          <svg
            class="c8 TEFileInputClearIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm0 394c0 3.3-2.7 6-6 6H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h404c3.3 0 6 2.7 6 6v340zM356.5 194.6L295.1 256l61.4 61.4c4.6 4.6 4.6 12.1 0 16.8l-22.3 22.3c-4.6 4.6-12.1 4.6-16.8 0L256 295.1l-61.4 61.4c-4.6 4.6-12.1 4.6-16.8 0l-22.3-22.3c-4.6-4.6-4.6-12.1 0-16.8l61.4-61.4-61.4-61.4c-4.6-4.6-4.6-12.1 0-16.8l22.3-22.3c4.6-4.6 12.1-4.6 16.8 0l61.4 61.4 61.4-61.4c4.6-4.6 12.1-4.6 16.8 0l22.3 22.3c4.7 4.6 4.7 12.1 0 16.8z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
